### ISSUE TITLE
Reconsider the use of generics in `DataTransfer`

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#145707]
Testing #145577

Todays `DataTransfer` is generic over the specific variant of the `DataTransferItem`. I wonder if that's a relict of an older variant of the API and what its use is. By now, the API is that `DataTransfer`-instances are passed into the `handleDrag/Drop` methods and therefore outside the control of extensions. 

We would still allow to set an subtype into a data transfer object but won't have it generic anymore, esp. as the changes are high that some [type, item] tuples are of different types

**Comment by alexr00** [comment]
Yes, this is a leftover. @mjbvz if you also have no need for the generic then I will remove it.

**Comment by mjbvz** [comment]
For drop into editor we only use a readonly data transfer so don't have any need for the types being generic
